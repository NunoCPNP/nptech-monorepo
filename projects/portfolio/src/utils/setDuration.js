export const setDuration = (minutes) => {
  const duration = new Date()

  duration.setTime(duration.getTime() + minutes * 60 * 1000)

  return duration
}
