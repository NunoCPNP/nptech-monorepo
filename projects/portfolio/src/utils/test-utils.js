import PropTypes from 'prop-types'
import { render } from '@testing-library/react'

import { ThemeProvider } from 'emotion-theming'
import { dark } from '../styles/themes'
import { ControllersProvider } from '../context/controllers'

// eslint-disable-next-line react/prop-types
const AllTheProviders = ({ children }) => {
  return (
    <ThemeProvider theme={dark}>
      <ControllersProvider>{children}</ControllersProvider>
    </ThemeProvider>
  )
}

const customRender = (ui, options) => render(ui, { wrapper: AllTheProviders, ...options })

// re-export everything
export * from '@testing-library/react'

// override render method
export { customRender as render }
