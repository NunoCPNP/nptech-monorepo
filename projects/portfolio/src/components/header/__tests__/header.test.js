import { render, cleanup } from '../../../utils/test-utils'

import Header from '../index'

afterEach(cleanup)

describe('<Header /> spec', () => {
  it('Should run without errors', () => {
    const { queryAllByTestId } = render(<Header />)

    const header = queryAllByTestId(/header/i)
    expect(header).toHaveLength(1)
  })

  it('Should render children navbar component', () => {
    const { queryAllByTestId } = render(<Header />)

    const navbar = queryAllByTestId(/navbar/i)
    expect(navbar).toHaveLength(1)
  })

  it('Should match snapshot', () => {
    const { asFragment } = render(<Header />)

    expect(asFragment(<Header />)).toMatchSnapshot()
  })
})
