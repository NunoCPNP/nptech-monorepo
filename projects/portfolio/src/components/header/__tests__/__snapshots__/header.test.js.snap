// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> spec Should match snapshot 1`] = `
<DocumentFragment>
  <header
    data-testid="header"
  >
    <nav
      class="css-15gg7lv"
      data-testid="navbar"
    >
      <div
        class="container"
      >
        <div
          class="navbar-logo"
        >
          <div />
        </div>
      </div>
    </nav>
  </header>
</DocumentFragment>
`;
